content/knowledge-codemagic/build-inputs.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Build inputs
+description: Specify optional inputs that are passed to the workflow when starting a build
+weight: 3
+---
+
+## Overview
+
+Build inputs are customizable parameters you can define within your workflow to make it more adaptable to different scenarios. With build inputs, you can create a single workflow and run it with different configurations by providing the values for inputs when starting a build for the workflow. For example, you can use build inputs to determine whether to build the workflow for test or release purposes or which app flavor to build. This eliminates the need to create multiple similar workflows with specific hardcoded values, making the workflow more reusable and dynamic.
+
+Inputs are workflow-specific and are defined in `codemagic.yaml` under the `inputs` mapping (see the [example](#minimal-example) below). The started workflow receives specified input values in the `inputs` context, i.e. `${{ inputs.inputId }}` is replaced with the value passed to input with identifier `inputId`.
+
+### Minimal example
+
+This example configures one input with the ID `name`. Unless given another value when starting a build, `name` defaults to `Codemagic`.
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  greetings:
+    inputs:
+      name:
+        description: Who is greeted?
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting builds with Inputs
+
+All inputs must be specified to successfully start a build, either by providing a `default` value in the YAML configuration, or giving a one-off value when starting the build. 
+
+{{<notebox>}}
+**Note**: Builds will fail if invalid values are provided (strings for number inputs, undefined choice options, etc.) or values are missing.
+{{</notebox>}}
+
+### Starting builds manually via Codemagic UI
+
+When starting a build via the Codemagic UI, you will automatically be prompted to enter the inputs. Inputs that have predefined default values will be prefilled with those values from configuration file. All other inputs must be manually entered before the build can be started.
+
+{{<notebox>}}
+**Note on inputs without default values:**
+
+Not entering anything for a string input will result in an empty string, i.e. `""`. Boolean inputs default to `false` in the UI. For other input types, an actual value that matches the requested type must be entered.
+{{</notebox>}}
+
+### Starting builds using webhook events
```

Comment:
the header doesn't reflect scheduled or other ways without UI interaction - see my suggest above